cudax/examples/stf/linear_algebra/cg_csr.cu

```diff
@@ -54,6 +60,7 @@ public:
   // Copy constructor
   vector(const vector& a)
   {
+    ctx    = a.ctx;
     handle = ctx.logical_data(a.handle.shape());
     copy_vector(a, *this);
   }
```

Comment:
```suggestion
  vector(const vector& a) : ctx(a.ctx), handle(ctx.logical_data(a.handle.shape()))
  {
    copy_vector(a, *this);
  }
```

I think. The idea is to initialize values with the right things from the get-go instead of initializing them with defaults and then assigning them.